### PR TITLE
[MNG-5222] Improvement in deprecated params detection

### PR DESCRIPTION
**Pull request by slawekjaranowski**
for https://github.com/apache/maven/pull/722

### CODE REVIEW

File: core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/DeprecatedConfigMojo.java [hunk 1]
```diff
@@ -47,6 +48,18 @@
...
+     * @deprecated bean read only
+     */
+    @Parameter(  defaultValue = "${project.artifact}", readonly = true )
+    private Artifact deprecatedBeanReadOnly;
+
+    /**
+     * @deprecated bean
+     */
+    @Parameter(  defaultValue = "${project.artifact}" )
+    private Artifact deprecatedBean;
```
Comment (michael-o) [inline core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/DeprecatedConfigMojo.java:61]: I don't see the corresponding test...

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5222MojoDeprecatedParamsTest.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
 {
     public MavenITmng5222MojoDeprecatedParamsTest()
     {
-        super( "[3.9.0,)" );
+        super( "[4.0.0-alpha-1,)" );
```
Comment (michael-o) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5222MojoDeprecatedParamsTest.java:38]: Why back to 4 and not with 3.9?

Comment (slawekjaranowski) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5222MojoDeprecatedParamsTest.java:38]: When this change will be merged with current fix to Maven master test on 3.9 will fail. 
I will restore it with together cherry pick/port change to 3.9.x branch of Maven core.

File: core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/DeprecatedConfigMojo.java (hunk 1 above)
Comment (slawekjaranowski) [reply core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/DeprecatedConfigMojo.java:61]: It is corner case when internal value, project object is used as property without read only flags.
Such parameter should not generated warning.

Comment (michael-o) [reply core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/DeprecatedConfigMojo.java:61]: I see, there is one space too much

**Pull request merged**